app/actions/utils.py

```diff
@@ -21,6 +25,12 @@ def generate_geometry_fragments(geometry_collection, interval=0.5):
         geometry_collection = geometry_collection.buffer(0)
 
     envelope = geometry_collection.envelope
+
+    # Check if envelope has valid bounds
+    if not envelope.bounds:
+        logger.warning(f"Geometry collection has no bounds: {geometry_collection}")
+        yield "error"
+
```

Comment:
We should raise an exception here. And this will let it bubble up to be logged as an error in activity log.

```suggestion

    # Check if envelope has valid bounds
    if not envelope.bounds:
        logger.warning(f"Geometry collection has no bounds: {geometry_collection}")
        raise ValueError("The geometry collection does not have valid envelope bounds.")


```